metrics/sources/summary/summary.go

```diff
@@ -194,10 +194,25 @@ func (this *summaryMetricsSource) decodePodStats(metrics map[string]*MetricSet,
 
 	for _, container := range pod.Containers {
 		key := PodContainerKey(ref.Namespace, ref.Name, container.Name)
+		// This check ensures that we are not replacing metrics of running container with metrics of terminated one if
+		// there are two exactly same containers reported by kubelet.
+		if _, exist := metrics[key]; exist && containerIsTerminated(&container, metrics[key].CreateTime) {
+			continue
```

Comment:
Thanks for the comment. I have tried to include all information in the log that could help to debug issue.

Could you check it once again?